app/src/main/java/com/google/android/stardroid/activities/util/OnSingleClickListener.java

```diff
@@ -0,0 +1,23 @@
+package com.google.android.stardroid.activities.util;
+
+import android.os.SystemClock;
+import android.view.View;
+
+public abstract class OnSingleClickListener implements View.OnClickListener {
+
+    private static final long MIN_CLICK_INTERVAL = 800;
```

Comment:
So I haven't looked into this too deeply, but as I understand it you're able to bring up the date picker modal dialog twice by tapping quickly?  Is this the right way to solve it? How do you know MIN_CLICK_INTERVAL is right?  Perhaps we should disable the button itself in the click listener and re-enable it when the modal closes?

Reply:
I thought about disabling a button when a diаlog opens. But when the diаlog opens, the datePicker and the timePicker button is overlaps. That is, it can not be clicked.

The problem is that in the time between the first pressing of a button and the opening of a dialog, it is possible to quickly press several times. It is then that the dialog opens several times. and I assumed that until the opening of the dialog after the first pressing MIN_CLICK_INTERVAL would be sufficient and decided to block all clicks in the interval smaller than the MIN_CLICK_INTERVAL

Reply:
Sorry - I'm still puzzled by this. Are you saying that you can't disable the button when it's clicked?

Reply:
no, its no disable button, it disables onClick event

> 
> 
> Sorry - I'm still puzzled by this. Are you saying that you can't disable the button when it's clicked?

no, its not disable button, its disable onClick event

Reply:
Yes, I see what it does. I was just asking if there's a reason why the button(s) _couldn't_ be disabled instead?

Reply:
i tested with disabling button and onClick, they are both worked nice. But when we disables button its color is changed to grey. You can see it here, on pickDate button disabled, on pickTime disables onClickevent:
![Screenrecorder201903302332](https://user-images.githubusercontent.com/28304184/55279684-6abb0100-5345-11e9-9b93-28f832f6c6e7.gif)